src/Options/Gravity.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Onliner\ImgProxy\Options;
+
+use InvalidArgumentException;
+use Onliner\ImgProxy\Support\GravityType;
+
+final class Gravity extends Option
+{
+    /**
+     * @var GravityType
+     */
+    private $type;
+    /**
+     * @var float|null
+     */
+    private $x;
+    /**
+     * @var float|null
+     */
+    private $y;
+
+    /**
+     * @param string $type
+     * @param mixed $x
+     * @param mixed $y
+     */
+    public function __construct(string $type, $x = null, $y = null)
+    {
+        $this->type = new GravityType($type);
+
+        if ($x !== null) {
+            if (!is_numeric($x) || ($x < 0 || $x > 1)) {
+                throw new InvalidArgumentException(sprintf('Invalid gravity X: %s', $x));
+            }
+
+            $this->x = (float) $x;
+        }
+
+        if ($y !== null) {
+            if (!is_numeric($y) || ($y < 0 || $y > 1)) {
+                throw new InvalidArgumentException(sprintf('Invalid gravity Y: %s', $y));
+            }
+
+            $this->y = (float) $y;
+        }
+    }
+
+    /**
+     * @param string $gravity
+     *
+     * @return static
+     */
+    public static function fromString(string $gravity): self
+    {
+        $params = explode(':', $gravity, 3);
+
+        return new self($params[0], $params[1] ?? null, $params[2] ?? null);
```

Comment:
все же, идея с ?float мне нравится.
```suggestion
        $x = isset($params[1]) && is_numeric($params[1]) ? (float) $params[1] : null;
        $y = isset($params[2]) && is_numeric($params[2]) ? (float) $params[2] : null;

        return new self($params[0], $x, $y);
```
необязательно тут делать проверки. просто раз уж это фабрика, то можно и кастануть

Reply:
в таком случае не будет обработки ошибок для not numeric данных.
например `ce:foo:bar` в таком случае занулится. я этого не хочу.

Reply:
я набросил вариант. конечно, можно сделать корректнее. но просто ты ввел фабричный метод, и под него ты упрощаешь типы аргументов в конструкторе. хотя, кмк, именно конструктор должен говорить, что тип `?float`, а значит фабрика подстраивается и делает проверки своей зоны ответственности